arrow/arrow.py

```diff
@@ -1,47 +1,63 @@
+
+from __future__ import absolute_import, division, print_function
+
+from itertools import izip
+
 import numpy as np
 
+from arrow.math import multichoose
 
-def choose(n, k):
-    terms = [
-        (n + 1.0 - i) / i
-        for i in xrange(1, k + 1)]
-    product = np.array(terms).prod()
-    return np.rint(product)
+def _get_reactants_and_stoichiometries(stoichiometric_matrix):
+    reactants = [
+        np.where(stoichiometry < 0)[0]
+        for stoichiometry in stoichiometric_matrix.T
+        ]
```

Comment:
This PR is great but I just can't get behind these dangling braces. PEP8 itself recommends keeping them on the same line as the one before. Do you have any other reason for preferring the dangling brace?

Reply:
Mmm, several reasons.  Symmetry, for one.  If the opening brace has no following characters on that line, then, likewise, I expect to see no leading characters on the line with the closing brace.  It's also easier to add new lines - e.g. in a list of items, I'm not constantly pushing the brace to a new line, it's already where it needs to be.  Further I find it easier to read and identify levels of precedence if the brace is given a new line.  Finally, if I have reason to comment out individual items (maybe each line is an element in a list, or an optional argument to a function), it's much easier to disable the last item if I don't then have to migrate a bracket.  I actually encountered specifically this issue during development - I wanted to run the test systems but only the first one until I finished debugging, but with the brace on the same line as the last element, toggling wasn't so easy:
https://github.com/CovertLab/arrow/blob/654a2692092368b09a22058ce3ef9c1977b59e2f/arrow/test/test_arrow.py#L120-L124

I could be missing it but I can't find an explicit recommendation one way or another in PEP 8.  In fact all I can find is the explicit optional choice of putting a trailing brace on either the same level of precedence as the items, or at the level below.  I've always had it at the same level simply because that's what Sublime Text does by default.

Reply:
Hmmm looking back through it I was basing this on all of the code examples they give where they keep the paren on the same line:

```
# Aligned with opening delimiter.
foo = long_function_name(var_one, var_two,
                         var_three, var_four)

# More indentation included to distinguish this from the rest.
def long_function_name(
        var_one, var_two, var_three,
        var_four):
    print(var_one)

# Hanging indents should add a level.
foo = long_function_name(
    var_one, var_two,
    var_three, var_four)

```

But it looks like they make a distinction between parens and braces? Which seems silly. Sigh.

Reply:
I'm not sure whether they are making a distinction, or being vague.